frontend/src/APIClients/SchedulingAPIClient.ts

```diff
@@ -25,14 +25,17 @@ const getScheduleById = async (scheduleId: string): Promise<Schedule> => {
   }
 };
 
-const getScheduleByDonorId = async (donorId: string): Promise<Schedule[]> => {
+const getScheduleByDonorId = async (
+  donorId: string,
+  weekLimit?: string,
+): Promise<Schedule[]> => {
   try {
-    const { data } = await baseAPIClient.get(
-      `/scheduling/?donorId=${donorId}`,
-      {
-        headers: { Authorization: BEARER_TOKEN },
-      },
-    );
+    const url = weekLimit
+      ? `/scheduling/?donorId=${donorId}&weekLimit=${weekLimit}`
+      : `/scheduling/?donorId=${donorId}`;
+    const { data } = await baseAPIClient.get(url, {
+      headers: { Authorization: BEARER_TOKEN },
+    });
```

Comment:
what about making `weekLimit` a mandatory parameter, but passing in `0` if there's no limit. This removes the ternary conditions in this file and `schedulingRoutes.ts` and I think is overall a bit cleaner.

Inside of the service we can just keep the same condition but check for `weekLimit > 0`